badius-osic-green-tribes/src/test/java/com/greenfox/exam/badiusosicgreentribes/transaction/handler/ProductionHandlerTest.java

```diff
@@ -0,0 +1,140 @@
+import com.greenfox.exam.badiusosicgreentribes.domain.battle.Troop;
+import com.greenfox.exam.badiusosicgreentribes.domain.battle.Unit;
+import com.greenfox.exam.badiusosicgreentribes.domain.battle.UnitType;
+import com.greenfox.exam.badiusosicgreentribes.domain.kingdom.BuildingType;
+import com.greenfox.exam.badiusosicgreentribes.domain.kingdom.Kingdom;
+import com.greenfox.exam.badiusosicgreentribes.domain.kingdom.Storage;
+import com.greenfox.exam.badiusosicgreentribes.domain.transaction.Production;
+import com.greenfox.exam.badiusosicgreentribes.domain.transaction.ProductionUnitType;
+import com.greenfox.exam.badiusosicgreentribes.domain.transaction.TransactionState;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class ProductionHandlerTest {
+
+    ProductionHandler handler = new ProductionHandler();
+
+    @Test
+    void confirm_CompletedTransaction() {
+        Kingdom kingdom = new Kingdom();
+        Storage storage = Storage.builder()
+                .gold(100)
+                .food(50)
+                .build();
+        kingdom.setStorage(storage);
+        Production transaction = Production.builder()
+                .productionType(ProductionUnitType.GOLD)
+                .quantity(10)
+                .targetKingdom(kingdom)
+                .state(TransactionState.COMPLETED)
+                .build();
+
+        handler.confirm(transaction);
+
+        assertEquals(110, storage.getGold());
+        assertEquals(50, storage.getFood());
+    }
+
+    @Test
+    void confirm_NotCompletedTransaction() {
+        Kingdom kingdom = new Kingdom();
+        Storage storage = Storage.builder()
+                .gold(100)
+                .food(50)
+                .build();
+        kingdom.setStorage(storage);
+        Production transaction = Production.builder()
+                .productionType(ProductionUnitType.GOLD)
+                .quantity(10)
+                .targetKingdom(kingdom)
+                .state(TransactionState.CANCELLED)
+                .build();
+
+        handler.confirm(transaction);
+
+        assertEquals(100, storage.getGold());
+        assertEquals(50, storage.getFood());
+    }
+
+    @Test
+    void confirm_CompletedTransaction_Farm() {
+        Kingdom kingdom = new Kingdom();
+        Storage storage = new Storage();
+        storage.setBuildings(new ArrayList<>());
+        kingdom.setStorage(storage);
+        Production transaction = Production.builder()
+                .productionType(ProductionUnitType.FARM)
+                .quantity(1)
+                .targetKingdom(kingdom)
+                .state(TransactionState.COMPLETED)
+                .build();
+
+        handler.confirm(transaction);
+
+        assertTrue(kingdom.getStorage().getBuildings().stream()
+                .anyMatch(building -> building.getType() == BuildingType.FARM));
+    }
```

Comment:
A transaction state-tel itt nincs semmi dolgunk. 
Nincs lefedve általa az össze építmény ami kieshet egy production-ből ( aka construction) . Ide is javasolok egy paraméterezett tesztet a következő argument listával: 

1. ProductionUnitType ( csak a building-ek)